internal/controller/cephfscg/volumegroupsourcehandler.go

```diff
@@ -285,13 +292,13 @@ func (h *volumeGroupSourceHandler) RestoreVolumesFromSnapshot(
 
 	volumeSnapshot := &vsv1.VolumeSnapshot{}
 	if err := h.Client.Get(ctx,
-		types.NamespacedName{Name: vsRef.Name, Namespace: vsRef.Namespace},
+		types.NamespacedName{Name: vsName, Namespace: pvc.Namespace},
```

Comment:
Maybe it's better to look for volume snapshot by vs name and vs namespace? What if one day vs namespace will be different from pvc namespace?

Reply:
The `VolumeGroupSnapshot` status looks like this:
```
status:
  boundVolumeGroupSnapshotContentName: groupsnapcontent-66e442b9-d739-48cc-9158-a6472c3a2022
  creationTime: "2024-09-13T13:05:00Z"
  pvcVolumeSnapshotRefList:
  - persistentVolumeClaimRef:
      name: busybox-cg2-pvc1
    volumeSnapshotRef:
      name: snapshot-bc3b0f7a8da345fd3d423eebe43cf6023ad34b03d7ff1df3fdf227da330303e6-2024-09-13-1.5.0
  - persistentVolumeClaimRef:
      name: busybox-cg2-pvc2
    volumeSnapshotRef:
      name: snapshot-811a028ea7008016c74c082430686955ae178d56cc4f0fdc3e5394da2cc20a1f-2024-09-13-1.5.3
  readyToUse: true
  ```
We have only the names in the status. The snapshot is for the PVC and we already have the PVC for it.  Therefore we use the PVC namespace.  The PVC namespace is the same as the `VolumeGroupSnapshot`.